File: ESTRUTURA DE DADOS II/Material/Códigos/Grafos/Busca-em-profundidade.py
def dfs(grafo, inicio, visitado=None):
    if visitado is None:
        visitado = [False] * len(grafo)
        
    visitado[inicio] = True
    print(f"Visitado {inicio}")
    
    for i, adjacente in enumerate(grafo[inicio]):
        if adjacente and not visitado[i]:
            dfs(grafo, i, visitado)

grafo = [[0, 1, 1, 0], [1, 0, 1, 1], [1, 1, 0, 1], [0, 1, 1, 0]]

dfs(grafo, 2) # Altere o segundo parâmetro!

File: ESTRUTURA DE DADOS II/Material/Códigos/Grafos/Busca-em-profundidade.c
#include <stdio.h>
#include <stdbool.h>

#define QUANTIDADE_VERTICES 4

void DFS(int grafo[QUANTIDADE_VERTICES][QUANTIDADE_VERTICES], bool visitado[QUANTIDADE_VERTICES], int vertice, int vertices) {
    visitado[vertice] = true;
    printf("Visitado %d\n", vertice);

    for (int i = 0; i < vertices; i++)
        if (grafo[vertice][i] && !visitado[i])
            DFS(grafo, visitado, i, vertices);
}

int main() {
    int grafo[QUANTIDADE_VERTICES][QUANTIDADE_VERTICES] = {{0, 1, 1, 0}, {1, 0, 1, 1}, {1, 1, 0, 1}, {0, 1, 1, 0}};
    bool visitado[QUANTIDADE_VERTICES] = {false};

    DFS(grafo, visitado, 1, QUANTIDADE_VERTICES); // Altere o terceiro parâmetro!

    return 0;
}